### DISCUSSION TITLE
Add antivirus scans for before file upload

### DISCUSSION THREAD

**Discussion by workatease** [directus/directus#4830] [Feature Requests]
I had a issue with the file upload that an virus was uploaded by a user embedded in an image.
was think if we could bring it antivirus scans  before uploading to the storage account.

Was think of using this open source antivirus. 
https://www.clamav.net/

Please suggest does this sounds like a good feature to add to Directus Core with options to configure it as needed.

**Comment by sabinehoi** [comment]
Hi, I am new web development and just curious, does it matter if a virus was uploaded? If we are never placing execute permissions on those files, does it matter what their content is?

I think we should have at least file size checking though

**Reply by workatease** [reply]
It not only for your server... It how safe are you making your application to be used by other users or a public facing application/website

Use case
- You created an website where users are allowed to upload pictures
- users machine is infected with a malware and upload an image to your server
-  you have allow public machines to download the image
- now any user in the world could get infected with the malware

This is just an example

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
